File: CourseMaterial/00_foundations_of_intelligence/02_chatbot_search_game/chatbot_output_improved/landscapeGame.js
const canvas = document.getElementById('landscapeCanvas');
const ctx = canvas.getContext('2d');
const guessCountElement = document.getElementById('guessCount');

const width = canvas.width;
const height = canvas.height;

let landscape = [];
let maxPeak = 0;
let guesses = 0;

// Function to generate a random landscape with mountains and valleys
function generateLandscape() {
    const peaks = 10; // Number of peaks in the landscape
    for (let i = 0; i < width; i++) {
        // Create height with random variation, simulating mountains and valleys
        landscape[i] = Math.sin(i * 0.02) * 100 + Math.random() * 10 + height / 2;
    }
    maxPeak = Math.min(...landscape); // Find the tallest peak
}

// Function to draw the landscape covered by black screen
function drawLandscape() {
    ctx.fillStyle = 'black';
    ctx.fillRect(0, 0, width, height);
}

// Function to reveal a vertical slice of the landscape where clicked
function revealSlice(x) {
    ctx.clearRect(x, 0, 1, height);
    ctx.beginPath();
    for (let i = -5; i < 5; i++) {
        ctx.moveTo(x + i, height);
        ctx.lineTo(x + i, landscape[x + i]);
        ctx.strokeStyle = 'green';
        ctx.stroke();
    }
}

// Function to handle clicks on the canvas
function handleClick(event) {
    const x = event.offsetX;
    revealSlice(x);

    guesses += 1;
    guessCountElement.textContent = guesses;

    // Check if the revealed slice is the tallest peak
    for (let i = -5; i < 5; i ++) {
        if (landscape[x + i] === maxPeak) {
            alert(`You've found the tallest peak in ${guesses} guesses!`);
            canvas.removeEventListener('click', handleClick); // Stop further clicks
        }
    }
}

// Initialize the game
function init() {
    generateLandscape();
    drawLandscape();
    canvas.addEventListener('click', handleClick);
}

init();

File: CourseMaterial/00_foundations_of_intelligence/05_skills_assessment_1/chatbot_output_python/landscape_game.py
import math
import random

from .screen import Context, get_click_location

ctx = Context()
width = ctx.width
height = ctx.height
landscape = []


def generate_landscape():
    """
    Function to generate a random landscape with mountains and valleys
    """
    for i in range(0, width):
        # Create height with random variation, simulating mountains and valleys
        landscape.append(math.sin(i * 0.02) * 100 + random.random() * 50 + height / 2)

    max_peak = max(landscape)  # Find the tallest peak
    return max_peak


def reveal_slice(x):
    """
    Function to reveal a vertical slice of the landscape where clicked
    """
    ctx.clear_rect(x, 0, 1, height)
    ctx.begin_path()
    for i in range(-5, 5):
        ctx.move_to(x + i, height)
        ctx.line_to(x + i, landscape[x + i])
        ctx.stroke()


def handle_click(event, max_peak):
    """
    Function to handle clicks on the canvas
    """
    x = event.offset_x
    reveal_slice(x)

    # Check if the revealed slice is the tallest peak
    for i in range(-5, 5):
        if landscape[x + i] == max_peak:
            return True

    return False


def init():
    """
    Function to initialize the game
    """
    max_peak = generate_landscape()

    guesses = 0
    found_peak = False
    while not found_peak:
        event = get_click_location()

        found_peak = handle_click(event, max_peak)
        if found_peak:
            # Stop further clicks
            print("You found the peak after", guesses, "guesses!")
        else:
            print("Try again!")
            guesses += 1


if __name__ == "__main__":
    init()
